### PR TITLE
Enable "Next button" on paginated view when nested answer is chosen

### LINKED ISSUE #1677: Inconsistencies when using the skipping pages and review functionality
**Describe the bug**
When using the skipping pages [functionality](https://github.com/google/android-fhir/pull/1532) together with the review [functionality](https://github.com/google/android-fhir/pull/1310), there are inconsistencies occurring. 

In this specific case, there is a questionnaire with multiple pages that are hidden by default. To enable the pages, a Boolean item needs to be answered. If the answer is 'no', then none of the remaining pages are enabled and the user proceeds to the review page. If the answer is 'yes', then the remaining pages are enabled and the user needs to go through those before seeing the review page. 

However, when the answer is 'yes', the review button remains and the user cannot proceed to the newly enabled pages. As a workaround, the user needs to tap the review button then tap edit button to return to the previous page where the next button is now available. Videos have been attached in the screenshots section to demonstrate this.

**Component**
SDC Library

**To Reproduce**
N/A

**Expected behavior**
I expect that when 'yes' is selected, the 'review' button changes to a 'next' button and allows the user to proceed to the newly enabled pages.

**Screenshots**
As per the above example, this is what happens when no is selected:
https://user-images.githubusercontent.com/90775548/196679194-79d51e6c-bf72-4dc2-9b93-3d7ab22c2bdf.mp4

As per the above example, this is what happens when yes is selected:
https://user-images.githubusercontent.com/90775548/196679514-12fb0b08-9d8d-456c-a63d-a74d2948548e.mp4

**Smartphone (please complete the following information):**
- Device: Samsung A12
- Android version: 11
- Build number: RP1A.200720.012.A127FXXU4AUK1
- Version: 30

**Additional context**
We are implementing this using [FHIR Core](https://github.com/opensrp/fhircore). The review page was implemented in this [PR](https://github.com/opensrp/fhircore/pull/1634).

Here is the questionnaire being used in the video:
```
"item": [
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
                    "valueCodeableConcept": {
                        "coding": [
                            {
                                "system": "http://hl7.org/fhir/questionnaire-item-control",
                                "code": "page",
                                "display": "Page 1"
                            }
                        ],
                        "text": "Page 1"
                    }
                }
            ],
            "linkId": "page-1",
            "type": "group",
            "item": [
                {
                    "linkId": "ict-consent-explanation",
                    "text": "We would like to offer HIV counselling and testing services to your loved ones or anyone in your social circle who is at risk of having HIV. Making sure your loved ones have access to HIV testing is a way for you to help keep them safe and healthy. By knowing their status, if they are HIV-positive they can access treatment and care early, and for those who are negative, they can access prevention services so they remain negative. There are different ways we can assist you to invite your loved ones (sexual partners, family members or any anyone who you consider to be at risk of having HIV) for counseling and HIV testing services. Is it possible for me to explain to you different ways we can assist you to help your loved ones access testing here or in the community? This process is completely voluntary. You can choose to decline at any point during our conversation.",
                    "type": "display"
                }
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
                    "valueCodeableConcept": {
                        "coding": [
                            {
                                "system": "http://hl7.org/fhir/questionnaire-item-control",
                                "code": "page",
                                "display": "Consent Given"
                            }
                        ],
                        "text": "Consent Given"
                    }
                }
            ],
            "linkId": "consent-page",
            "text": "<h1>ICT Information and Consent</h1>",
            "type": "group",
            "item": [
                {
                    "linkId": "ict-consent",
                    "text": "<b>ASK:</b> Do you give consent to conduct ICT?",
                    "type": "boolean",
                    "required": true
                }
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
                    "valueCodeableConcept": {
                        "coding": [
                            {
                                "system": "http://hl7.org/fhir/questionnaire-item-control",
                                "code": "page",
                                "display": "Clients Contacts"
                            }
                        ],
                        "text": "Clients Contacts"
                    }
                }
            ],
            "linkId": "client-Contacts-page",
            "text": "<h1>Referral Methods and Contact Identification</h1>",
            "type": "group",
            "enableWhen": [
                {
                    "question": "ict-consent",
                    "operator": "=",
                    "answerBoolean": true
                }
            ],
            "enableBehavior": "all",
            "item": [
                {
                    "linkId": "explain-client",
                    "text": "<b>EXPLAIN</b> to the client that we have 4 referral approches which include the following: You inform your contact to come for testing within 2 weeks. If the contact does not come, CHW will follow up with your contact directly With your consent, a CHW will directly follow up with the contact now A CHW will sit with you and your contact and together provides support to your contact Alternatively, we can give you an FRS to give each of your contacts inviting them to come for health services.",
                    "type": "display"
                },
                {
                    "linkId": "share-hiv-info-contacts",
                    "text": "<b>ASK:</b> Is it ok with you if we talk about HIV testing for each of your children, spouse and anyone in your social circle?",
                    "type": "boolean",
                    "required": true
                },
                {
                    "linkId": "contact-type",
                    "text": "Who are the client's contacts?",
                    "type": "choice",
                    "enableWhen": [
                        {
                            "question": "share-hiv-info-contacts",
                            "operator": "=",
                            "answerBoolean": true
                        }
                    ],
                    "enableBehavior": "all",
                    "required": true,
                    "repeats": true,
                    "answerOption": [
                        {
                            "valueCoding": {
                                "code": "child-contacts",
                                "display": "Child Contacts"
                            }
                        },
                        {
                            "valueCoding": {
                                "code": "sexual-contacts",
                                "display": "Sexual Contacts"
                            }
                        },
                        {
                            "extension": [
                                {
                                    "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-optionExclusive",
                                    "valueBoolean": true
                                }
                            ],
                            "valueCoding": {
                                "code": "no-contacts",
                                "display": "No Contacts"
                            }
                        }
                    ]
                }
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
                    "valueCodeableConcept": {
                        "coding": [
                            {
                                "system": "http://hl7.org/fhir/questionnaire-item-control",
                                "code": "page",
                                "display": "Child Contacts"
                            }
                        ],
                        "text": "Child Contacts"
                    }
                }
            ],
            "linkId": "child-contacts-page",
            "text": "<h1>Child Contacts</h1>",
            "type": "group",
            "enableWhen": [
                {
                    "question": "ict-consent",
                    "operator": "=",
                    "answerBoolean": true
                },
                {
                    "question": "contact-type",
                    "operator": "=",
                    "answerCoding": {
                        "code": "child-contacts",
                        "display": "Child Contacts"
                    }
                }
            ],
            "enableBehavior": "all",
            "item": [
                {
                    "linkId": "child-contacts-age-range-info",
                    "text": "<b>TELL</b> the client that child contacts are persons who are 0 - 19 years old",
                    "type": "display"
                },
                {
                    "linkId": "child-contacts-count",
                    "text": "<b>ENTER</b> the number of child contacts the client has",
                    "type": "integer",
                    "required": true
                },
                {
                    "extension": [
                        {
                            "url": "http://hl7.org/fhir/StructureDefinition/maxValue",
                            "valueInteger": 10
                        }
                    ],
                    "linkId": "child-contacts-count-untested",
                    "text": "<b>ENTER</b> the number of child contacts that have NOT been tested for HIV",
                    "type": "integer",
                    "required": true
                },
                {
                    "linkId": "finish-child-contacts-registration",
                    "text": "Register the Child Contacts in the Child Contacts Registration task after saving.",
                    "type": "display"
                }
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
                    "valueCodeableConcept": {
                        "coding": [
                            {
                                "system": "http://hl7.org/fhir/questionnaire-item-control",
                                "code": "page",
                                "display": "Sexual Contacts"
                            }
                        ],
                        "text": "Sexual Contacts"
                    }
                }
            ],
            "linkId": "page-sexual-contacts",
            "text": "<h1>Sexual Contacts</h1>",
            "type": "group",
            "enableWhen": [
                {
                    "question": "ict-consent",
                    "operator": "=",
                    "answerBoolean": true
                },
                {
                    "question": "contact-type",
                    "operator": "=",
                    "answerCoding": {
                        "code": "sexual-contacts",
                        "display": "Sexual Contacts"
                    }
                }
            ],
            "enableBehavior": "all",
            "item": [
                {
                    "linkId": "sexual-contacts-count",
                    "text": "<b>ENTER</b> the number of sexual contacts the client has",
                    "type": "integer",
                    "required": true
                },
                {
                    "extension": [
                        {
                            "url": "http://hl7.org/fhir/StructureDefinition/maxValue",
                            "valueInteger": 10
                        }
                    ],
                    "linkId": "sexual-contacts-count-untested",
                    "text": "<b>ENTER</b> the number of sexual contacts that have NOT been tested for HIV",
                    "type": "integer",
                    "required": true
                },
                {
                    "linkId": "finish-sexual-contact-registration",
                    "text": "Register the Sexual Contacts in the Sexual Contacts Registration task after saving [Add this task to the task list when saving]",
                    "type": "display"
                }
            ]
        }
    ]
```

### PR DESCRIPTION
**Pull request by omarismail94**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1677 : Enable "Next button" on paginated view when nested answer is chosen

**Description**
Run check whether to display review and submit button _after_ pages has been updated when pagination is on

**Type**
Choose one: (**Bug fix** | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**
[device-2022-10-21-142233.webm](https://user-images.githubusercontent.com/44980219/197218932-843cae12-1b9b-4bcb-b9a5-fbe2a26744f2.webm)

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by jingtang10 — CHANGES_REQUESTED** [review]
can we add some tests?

**Comment by omarismail94** [comment]
> can we add some tests?

Done! I added a UI test that loads a questionnaire, then clicks on the buttons to test if the `Review` button is visible

File: datacapture/sampledata/paginated_questionnaire_with_dependent_answer.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                "code": "page",
+                "display": "Page 1"
+              }
+            ],
+            "text": "Page 1"
+          }
+        }
+      ],
+      "linkId": "page-1",
+      "type": "group",
```
Comment (jingtang10) [inline datacapture/sampledata/paginated_questionnaire_with_dependent_answer.json:23]: i don't think we need this group at all to be honest. this doesn't contribute to testing what we want to test. and just makes the test slower.

**Review by jingtang10 — APPROVED** [review]

File: datacapture/sampledata/paginated_questionnaire_with_dependent_answer.json (hunk 1 above)
Comment (omarismail94) [reply datacapture/sampledata/paginated_questionnaire_with_dependent_answer.json:23]: Removed!

**Pull request merged**